incubator/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -3203,7 +3255,17 @@ private void onSessionBegin(
                 state = MqttState.openReply(state);
 
                 final long traceId = begin.traceId();
-                final long authorization = begin.authorization();
+
+                final OctetsFW extension = begin.extension();
+                if (extension.sizeof() > 0)
+                {
+                    final MqttBeginExFW mqttBeginEx = extension.get(mqttBeginExRO::tryWrap);
+
+                    assert mqttBeginEx.kind() == MqttBeginExFW.KIND_SESSION;
+                    final MqttSessionBeginExFW mqttSessionBeginEx = mqttBeginEx.session();
+
+                    sessionExpiryIntervalLimit = mqttSessionBeginEx.expiry();
```

Comment:
This is just the `sessionExpiry`, right? Do we need to track it as a limit separately for some reason?

Reply:
Nope, fixed it